account_maturity_date_default/models/account_move_line.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if not vals.get("date_maturity"):
+                account = self.env["account.account"].browse(vals.get("account_id"))
+                if account.internal_type in ["receivable", "payable"]:
```

Comment:
Not that it will make a big difference here, but in situations like these you can use sets instead of other Python structures, they will be significantly faster when it comes to determining if an object is present or not:
```suggestion
                if account.internal_type in {"receivable", "payable"}:
```
The same would not apply if you were iterating (i. e. `for i in ...`), as lists would be faster there...